buffer/src/main/java/io/netty5/buffer/api/ComponentIterator.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2022 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty5.buffer.api;
+
+import io.netty5.buffer.api.ComponentIterator.Next;
+
+public interface ComponentIterator<T extends Next> extends AutoCloseable {
```

Comment:
I wonder why we cant just use `MyIterator<T> extends, Iterator<T>, AutoCloseable` ? Is it because you want to ensure the user always close ?

Reply:
I'm trying to avoid allocations for single-component buffers, but maybe that's still possible with the JDK iterator interface.

Reply:
just if the iterator is cached and reused on close I think

Reply:
Not sure, depends if between allocation and iterator close there are operations which the JVM cannot figure it out what will happen next - meaning that it cannot mark the object as won't escape.
I can write a jmh bench to check it out btw, If it can help and see if it allocate anything